Assets/Character/Scripts/PlayerStateController.cs

```diff
@@ -0,0 +1,168 @@
+﻿using UnityEngine;
+
+public class PlayerStateController : MonoBehaviour
+{
+	public Animator animator;
+
+	private SpriteRenderer _spriteRenderer;
+	private bool _facingRight;
+	float _distToGround;
+
+	void Start ()
+	{
+		animator = this.GetComponent<Animator>();
+		animator.SetInteger("state", Constants.STATE_IDLE);
+		_facingRight = true;
+		_spriteRenderer = this.GetComponent<SpriteRenderer>();
+		_distToGround = this.GetComponent<Collider2D>().bounds.extents.y;
+	}
+
+	void Update()
+	{
+		if (InState(Constants.STATE_IDLE))
+		{
+			IdleStateUpdate();
+		}
+		else if (InState(Constants.STATE_MID_ATTACK))
+		{
+			MidAttackStateUpdate();
+		}
+		else if (InState(Constants.STATE_MOVE_LEFT) || InState(Constants.STATE_MOVE_RIGHT))
+		{
+			MoveStateUpdate();
+		}
+		else if (InState(Constants.STATE_ROLL))
+		{
+			RollStateUpdate();
+		}
+		else if (InState(Constants.STATE_JUMP))
+		{
+			JumpStateUpdate();
+		}
+		else if (FinishedCurrentAnimation())
+		{
+			animator.SetInteger("state", Constants.STATE_IDLE);
+		}
+	}
+
+	public bool InState(int state)
+	{
+		return animator.GetInteger("state") == state;
+	}
+
+	void ChangeState(int newState)
+	{
+		if (!InState(newState))
+		{
+			animator.SetInteger("state", newState);
+		}
+	}
+
+	bool PressedRoll()
+	{
+		return Input.GetKeyDown(KeyCode.Space);
+	}
+
+	bool PressedMidAttack()
+	{
+		return Input.GetKeyDown(KeyCode.A);
+	}
+
+	bool PressedJump()
+	{
+		return Input.GetKey(KeyCode.B);
+	}
+	bool PressedMoveLeft()
+	{
+		return Input.GetKeyDown(KeyCode.LeftArrow);
+	}
+	bool PressedMoveRight()
+	{
+		return Input.GetKeyDown(KeyCode.RightArrow);
+	}
+
+	void IdleStateUpdate()
+	{
+		if (!IsGrounded())
+		{
+			return;
+		}
+

```

Comment:
spriteflip = !facing_right 

put this in one place instead of repeating it or doing ternary